packages/earth-map/src/sagas/location/index.ts

```diff
@@ -99,6 +99,14 @@ function* toLocation({ payload, meta }) {
   } catch (e) {
     // TODO better error handling for sagas
     if ([403, 404].includes(e.request.status)) {
+      const {
+        global: { lastViewedPlace },
+      } = yield select(getAll);
+
+      if (lastViewedPlace?.slug === slug) {
+        yield put(setLastViewedPlace(null));
```

Comment:
You should probably `yield put(persistData());`to also change the session storage